test/AppLifecycle/FunctionalTests.cpp

```diff
@@ -0,0 +1,217 @@
+            WriteContentFile(c_testDataFileName_Packaged);
+
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            // Swallow errors in cleanup.
+            try
+            {
+                DeleteContentFile(c_testDataFileName_Packaged);
+                DeleteContentFile(c_testDataFileName);
+                UninstallPackage(c_testPackageFullName);
+                RunCertUtil(c_testPackageCertFile, true);
+            }
+            catch (const std::exception&)
+            {
+            }
+            catch (const winrt::hresult_error&)
+            {
+            }
+
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            m_failed = CreateTestEvent(c_testFailureEventName);
+            return true;
+        }
+
+        // Validate that UWP is not a supported scenario.
+        TEST_METHOD(GetActivatedEventArgsIsNull)
+        {
+            BEGIN_TEST_METHOD_PROPERTIES()
+                TEST_METHOD_PROPERTY(L"RunAs", L"UAP")
+            END_TEST_METHOD_PROPERTIES();
+
+            VERIFY_IS_NULL(AppLifecycle::GetActivatedEventArgs());
+        }
+
+        TEST_METHOD(GetActivatedEventArgsIsNotNull)
+        {
+            BEGIN_TEST_METHOD_PROPERTIES()
+                // Run this test for both PackagedWin32 and Win32.
+                TEST_METHOD_PROPERTY(L"RunAs", L"{UAP,InteractiveUser}")
+
+                // UAP:Host/UAP:AppXManifest are ignored when RunAs != UAP.
+                TEST_METHOD_PROPERTY(L"UAP:Host", L"PackagedCwa")
+                TEST_METHOD_PROPERTY(L"UAP:AppXManifest", L"PackagedCwaFullTrust")
+            END_TEST_METHOD_PROPERTIES();
+
+            VERIFY_IS_NOT_NULL(AppLifecycle::GetActivatedEventArgs());
+        }
+
+        TEST_METHOD(GetActivatedEventArgsForLaunch)
+        {
+            BEGIN_TEST_METHOD_PROPERTIES()
+                // Run this test for both PackagedWin32 and Win32.
+                TEST_METHOD_PROPERTY(L"RunAs", L"{UAP,InteractiveUser}")
+
+                // UAP:Host/UAP:AppXManifest are ignored when RunAs != UAP.
+                TEST_METHOD_PROPERTY(L"UAP:Host", L"PackagedCwa")
+                TEST_METHOD_PROPERTY(L"UAP:AppXManifest", L"PackagedCwaFullTrust")
+            END_TEST_METHOD_PROPERTIES();
+
+            auto args = AppLifecycle::GetActivatedEventArgs();
+            VERIFY_IS_NOT_NULL(args);
+            VERIFY_ARE_EQUAL(args.Kind(), ActivationKind::Launch);
+
+            auto launchArgs = args.as<LaunchActivatedEventArgs>();
+            VERIFY_IS_NOT_NULL(launchArgs);
+        }
+
+        TEST_METHOD(GetActivatedEventArgsForProtocol_Win32)
+        {
+            // Create a named event for communicating with test app.
+            auto event = CreateTestEvent(c_testProtocolPhaseEventName);
+
+            // Launch the test app to register for protocol launches.
+            if (!Execute(L"AppLifecycleTestApp.exe", L"/RegisterProtocol", g_deploymentDir))
```

Comment:
nit: May be just do VERIFY_NOT_NULL(Execute(...));? There is already a GetLastError in Execute() to get error if ShellExecute fails.

Reply:
Good catch!  I wrote these before updating the wrapper for the cert util calls, and forgot to update all the callers.